snapshot-labs/sx-monorepo
# fix: use the new `autoLogin` function to silently login users

### Summary



Toward https://github.com/snapshot-labs/sx-monorepo/issues/478
Depends on https://github.com/snapshot-labs/lock/pull/121

This PR will use the new `autoLogin` function in the lock library, which will log in a previously logged user silently, without triggering any wallet prompt

For now, this PR is only fixing the issue with the injected (metamask) wallet, argent X fix coming in another PR

### How to test

1. yarn link a local version of lock, on the branch `feat-add-auto-connect-function`
2. `yarn dev`
3. You can log in and log out as usual
4. When logged in, refreshing the page will log you in automatically
5. When metamask wallet is locked, visiting the site will not open the metamask unlock prompt anymore
6. Clicking on "Log in" > "Metamask" should prompt to unlock metamask wallet
